onnxscript/version_converter/version_converter.py

```diff
@@ -1,55 +1,427 @@
+import onnxscript.rewriter.pattern as orp
 from onnxscript import ir
-from onnxscript.version_converter._adapter_lib import pick_adapter_set
+
+logger = logging.getLogger(__name__)
+
+
+_ADAPTERS_18_19 = {
+    "Equal",
+    "AveragePool",
+    "Cast",
+    "CastLike",
+    "Constant",
+    "DequantizeLinear",
+    "Identity",
+    "If",
+    "Loop",
+    "Pad",
+    "QuantizeLinear",
+    "Reshape",
+    "Scan",
+    "Shape",
+    "Size",
+}
+
+
+_ADAPTERS_19_20 = {
+    "DFT",
+    "ConstantOfShape",
+    "IsInf",
+    "IsNan",
+    "ReduceMax",
+    "ReduceMin",
+    "GridSample",
+}
+
+
+_ADAPTERS_20_21 = {
+    "Cast",
+    "CastLike",
+    "Constant",
+    "ConstantOfShape",
+    "DequantizeLinear",
+    "Flatten",
+    "GroupNormalization",
+    "Identity",
+    "If",
+    "Loop",
+    "Pad",
+    "QLinearMatmul",
+    "QuantizeLinear",
+    "Reshape",
+    "Scan",
+    "Shape",
+    "Size",
+    "Squeeze",
+    "Transpose",
+    "Unsqueeze",
+}
+
+
+_ADAPTERS_21_22 = {
+    "EyeLike",
+    "RandomUniform",
+    "RandomNormal",
+    "RandomUniformLike",
+    "RandomNormalLike",
+    "Multinomial",
+    "Bernoulli",
+    "ThresholdedRelu",
+    "Selu",
+    "Elu",
+    "Mish",
+    "HardSigmoid",
+    "HardSwish",
+    "Softsign",
+    "Softplus",
+    "Sin",
+    "Cos",
+    "Tan",
+    "Asin",
+    "Acos",
+    "Atan",
+    "Sinh",
+    "Cosh",
+    "Asinh",
+    "Acosh",
+    "Atanh",
+    "Round",
+    "Det",
+    "NegativeLogLikelihoodLoss",
+    "AveragePool",
+    "MaxPool",
+    "MaxUnpool",
+    "LpPool",
+    "MaxRoiPool",
+    "Conv",
+    "ConvTranspose",
+    "DeformConv",
+    "GlobalAveragePool",
+    "GlobalMaxPool",
+    "GlobalLpPool",
+    "InstanceNormalization",
+    "LpNormalization",
+    "Dropout",
+    "RoiAlign",
+    "RNN",
+    "GRU",
+    "LSTM",
+    "GridSample",
+}
+
+
+_ADAPTER_SETS = {
+    (18, 19): _ADAPTERS_18_19,
+    (19, 20): _ADAPTERS_19_20,
+    (20, 21): _ADAPTERS_20_21,
+    (21, 22): _ADAPTERS_21_22,
+}
+
+
+@dataclasses.dataclass
+class Replacement:
+    """A replacement for a node in the graph."""
+
+    new_outputs: Sequence[ir.Value]
+    new_nodes: Sequence[ir.Node]
+
+
+# A partial-evaluator function takes a node, a RewriterContext, OptimizerState and returns
+# a Replacement for the node or None (if no replacement is needed). It may also return just
+# the ir.Value or ir.Values to replace the output values of the node, when the new nodes
+# can be inferred from the RewriterContext used to build the new nodes.
+
+ReturnValue = Union[Replacement, Sequence[ir.Value], ir.Value, None]
+AdapterFunction = Callable[[ir.Node, orp.RewriterContext], ReturnValue]
+
+
+@dataclasses.dataclass
+class AdapterVersionChecker:
+    """A class that represents a version checker for a particular op.
+
+    It is applicable for a specific version upgrade (orignal_version -> target_version) of the op.
+    """
+
+    node_version: int | None
+    upgrade_version: int | None
```

Comment:
minor nit: I am not sure if we need two ints. In future, we will likely add down-conversion, but I think we just need up or down conversion by one? Eg., 18 to 19 as well as 18 to 17. So, probably, a version-number (18) and a boolean (to indicate if this is for up-conversion or down-conversion) might be enough.